dot/sync/service.go

```diff
@@ -88,12 +89,27 @@ type Change struct {
 type Strategy interface {
 	OnBlockAnnounce(from peer.ID, msg *network.BlockAnnounceMessage) (repChange *Change, err error)
 	OnBlockAnnounceHandshake(from peer.ID, msg *network.BlockAnnounceHandshake) error
-	NextActions() ([]*SyncTask, error)
-	Process(results []*SyncTaskResult) (done bool, repChanges []Change, blocks []peer.ID, err error)
+	NextActions() ([]Task, error)
+	Process(results <-chan TaskResult) (done bool, repChanges []Change, blocks []peer.ID, err error)
 	ShowMetrics()
 	IsSynced() bool
 }
 
+type syncTask struct {
```

Comment:
Could you please move this to `fullsync.go` since it is specific for that strategy?